[MISC] deprecate seqan3::alignment_coordinate

SGSSGene opened the pull request:
Part of https://github.com/seqan/product_backlog/issues/282

SGSSGene commented:
Open: many other functions use `seqan3::alignment_coordinate` and probably have to be freed from it?

MitraDarja reviewed (approved):
LGTM. But can you check, what the problem with the CI is?

Irallia commented on test/unit/alignment/pairwise/fixture/alignment_fixture.hpp:
```diff
@@ -121,12 +124,15 @@ alignment_fixture(
 -> alignment_fixture<sequence1_t, sequence2_t, config_t, score_t,
                      std::vector<score_t>, std::vector<seqan3::detail::trace_directions>>;
 
+
```
Multiple new lines

Irallia commented on test/unit/alignment/pairwise/fixture/alignment_fixture.hpp:
```diff
@@ -62,8 +61,10 @@ struct alignment_fixture
 
-    seqan3::alignment_coordinate begin_positions;
-    seqan3::alignment_coordinate end_positions;
+    size_t begin1_position;
+    size_t begin2_position;
+    size_t end1_position;
+    size_t end2_position;
```
Maybe give the variables clearer names:
```suggestion
    size_t begin_column_position;
    size_t begin_row_position;
    size_t end_column_position;
    size_t end_row_position;
```

Irallia commented on test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp:
```diff
@@ -164,7 +164,8 @@ TYPED_TEST_P(edit_distance_unbanded_test, end_positions)
     auto end_positions = alignment.end_positions();
 
-    EXPECT_EQ(end_positions, fixture.end_positions);
+    EXPECT_EQ(end_positions.first, fixture.end1_position);
+    EXPECT_EQ(end_positions.second, fixture.end2_position);
```
```suggestion
    EXPECT_EQ(end_positions.first, fixture.end_column_position);
    EXPECT_EQ(end_positions.second, fixture.end_row_position);
```
or
```suggestion
    EXPECT_EQ(end_positions.first, fixture.end_first_position);
    EXPECT_EQ(end_positions.second, fixture.end_second_position);
```
?

marehr replied:
They are called

* sequence1_begin_position
* sequence1_end_position
* sequence2_begin_position
* sequence2_end_position

https://docs.seqan.de/seqan/3-master-user/classseqan3_1_1alignment__result.html#a6f33588c817b822d0ebe7068d3072b6c

Irallia commented on test/unit/alignment/pairwise/fixture/local_affine_banded.hpp:
```diff
@@ -52,8 +52,11 @@ static auto dna4_01 = []()
         "GTCTA",
-        seqan3::alignment_coordinate{column_index_type{5u}, row_index_type{2u}},
-        seqan3::alignment_coordinate{column_index_type{10u}, row_index_type{7u}},
+        // begin and end positions
```
You could add this comment in the other tests aswell 
eg `global_affine_unbanded.hpp`

Irallia reviewed (changes requested):
Looks very good already. I would only rename begin1, begin2, end1, end2.
Also, I like your comment, but you are just using it very randomly.
```cpp
 // begin and end positions
        0u,
        0u,
        22u,
        12u,
        ```

marehr commented on test/unit/alignment/pairwise/fixture/global_affine_banded.hpp:
```diff
@@ -55,8 +55,10 @@ static auto dna4_01 = []()
         "ACGTAC----------GTA",
-        seqan3::alignment_coordinate{seqan3::detail::column_index_type{0u}, seqan3::detail::row_index_type{0u}},
-        seqan3::alignment_coordinate{seqan3::detail::column_index_type{16u}, seqan3::detail::row_index_type{9u}},
+        0u,
+        0u,
+        16u,
+        9u,
```
Maybe you can reapply your regex and add

```suggestion
        /*.sequence1_begin_position = */ 0u,
        /*.sequence2_begin_position = */ 0u,
        /*.sequence1_end_position = */ 16u,
        /*.sequence2_end_position = */ 9u,
```

SGSSGene commented:
Very good suggestions! I adjusted my regex so it actually works correctly...changed it to `.sequence1_begin_position` also added the suggested comments.

Irallia reviewed (approved):
Perfect, thank you!
